Philips.CodeAnalysis.Common/TypesHelper.cs

```diff
@@ -49,5 +50,15 @@ public bool IsUserControl(INamedTypeSymbol type)
 		{
 			return IsInheritingFromClass(type, @"ContainerControl");
 		}
+
+		/// <summary>
+		/// Checks if the given type declaration is nested within another type
+		/// </summary>
```

Comment:
See following comment. I think with a re-design (rather than port from the other analyzer), this gets a lot simpler.